rts/Game/UI/SelectionKeyHandler.cpp

```diff
@@ -416,6 +422,32 @@ void CSelectionKeyHandler::DoSelection(std::string selectString)
 		}
 	}
 
+	if (!idMatchesSet.empty()) {
```

Comment:
```suggestion
	/* Positive IdMatches use OR instead of AND,
	 * because that is intuitive and it's not possible
	 * for a unit to match two names anyway. */
	if (!idMatchesSet.empty()) {
```